More descriptive message when random PARAFAC2 rank is infeasible given shape

#### Describe the bug
 
When Testing the random parafrac2 function fails throwing a error  " ValueError: All the projection matrices of a PARAFAC2 tensor should have the same number of columns as the rank. However, rank=2 but projections[0].shape[1]=1" while validating when parafrac2Tensor object is being created by projection_matrices, weights and factors.
 

#### Steps or Code to Reproduce

For instance:
Arguments passed to random Tensor are 
shapes = [(1, 1), (1, 1)]
rank =2  


#### Expected behavior
projection shape should have the same rank as rank passed both 2 
#### Actual result
But  getting projection rank as 1 and actual rank is 2 
Please paste or specifically describe the actual output or traceback. 
```
for i, projection in enumerate(projections):
            current_mode_size, current_rank = ivy.shape(projection)
            if current_rank != rank:
>               raise ValueError(
                    "All the projection matrices of a PARAFAC2 tensor should have the"
                    f" same number of columns as the rank. However, rank={rank} but"
                    f" projections[{i}].shape[1]={ivy.shape(projection)[1]}"
                )
E               ValueError: All the projection matrices of a PARAFAC2 tensor should have the same number of columns as the rank. However, rank=2 but projections[0].shape[1]=1

ivy/data_classes/factorized_tensor/parafac2_tensor.py:166: ValueError
```



#### Versions
<!--
Please run the following snippet and paste the output below.

--- a/tensorly/decomposition/_parafac2.py
+++ b/tensorly/decomposition/_parafac2.py
@@ -458,6 +458,15 @@ def parafac2(
     non-negative, then :math:`B` will be non-negative, but not the orthogonal `P_i` matrices.
     Consequently, the `B_i` matrices are unlikely to be non-negative.
     """
+    assert (
+        rank <= tensor_slices[0].shape[1]
+    ), f"PARAFAC2 rank ({rank}) cannot be greater than the number of columns in each tensor slice ({tensor_slices[0].shape[1]})."
+
+    for ii in range(1, len(tensor_slices)):
+        assert (
+            tensor_slices[0].shape[1] == tensor_slices[ii].shape[1]
+        ), "All tensor slices must have the same number of columns."
+
     weights, factors, projections = initialize_decomposition(
         tensor_slices, rank, init=init, svd=svd, random_state=random_state
     )

